### PR TITLE
feat: initial version of azure developer signing dataplane library

### PR DESCRIPTION
**Pull request by Jaxelr**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

Created using the typespec design created on this [pull request](https://github.com/Azure/azure-rest-api-specs/pull/26634)

cc: @DawnWang17 @janielmartell

### CODE REVIEW

File: sdk/developersigning/Azure.Developer.Signing/tests/DevSigningClientTests.cs
Comment (jsquire) [inline sdk/developersigning/Azure.Developer.Signing/tests/DevSigningClientTests.cs:14]: There does not seem to be a test resources file to accompany this; we'll need one of those so that tests can be recorded and run in Live mode.

File: sdk/developersigning/CONTRIBUTING.md
Comment (jsquire) [inline sdk/developersigning/CONTRIBUTING.md:2]: This doesn't seem to add any additional information to the repository contributing docs.   We should remove it.

**Review by jsquire — CHANGES_REQUESTED** [review]

**Comment by jsquire** [comment]
Service label for `Code Signing` has been added to the common set and pushed out to all managed repositories.   Let's use that when setting up CODEOWNERS.

File: sdk/developersigning/Azure.Developer.Signing/README.md
Comment (Jaxelr) [reply sdk/developersigning/Azure.Developer.Signing/README.md:93]: Good observation, I'm adding the full name and EKUs in parenthesis.

File: sdk/developersigning/Azure.Developer.Signing/src/GlobalSuppressions.cs
Comment (Jaxelr) [reply sdk/developersigning/Azure.Developer.Signing/src/GlobalSuppressions.cs:6]: There was indeed an archboard review that suggested the azure devloper signing namespace.

Comment (Jaxelr) [reply sdk/developersigning/Azure.Developer.Signing/src/GlobalSuppressions.cs:6]: You can also see the original typespec definition created on this [PR here](https://github.com/Azure/azure-rest-api-specs/pull/26634)

Comment (jsquire) [reply sdk/developersigning/Azure.Developer.Signing/src/GlobalSuppressions.cs:6]: Thanks!   I'll get the analyzer updated so that the suppression isn't needed.

File: sdk/developersigning/Azure.Developer.Signing/tests/DevSigningClientTests.cs
Comment (jsquire) [reply sdk/developersigning/Azure.Developer.Signing/tests/DevSigningClientTests.cs:14]: This looks to be the only thing left that is blocking.   See:
https://github.com/Azure/azure-sdk-for-net/blob/main/eng/common/TestResources/README.md

Comment (Jaxelr) [reply sdk/developersigning/Azure.Developer.Signing/tests/DevSigningClientTests.cs:14]: Thanks, I was running into some challenges with the configuration, but finally just pushed the change on the repo with the test-resources.bicep

**Comment by jsquire** [comment]
@Jaxelr:   Your account is missing write permissions to the repository, and we still need a CODEOWNERS entry for this.  Instructions and support can be found here:
- [Azure SDK onboarding](https://eng.ms/docs/products/azure-developer-experience/onboard/requirements) _(Microsoft internal)_
- [Azure SDK onboarding assistance](https://teams.microsoft.com/l/channel/19%3ade3fce22c2994be18cac50502c55f717%40thread.skype/Azure%2520SDK%2520Onboarding?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) _(Microsoft internal)_

**Comment by jsquire** [comment]
I overlooked that you're also missing a [tests.yml](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/appconfiguration/tests.yml) file for use with the live test runs.

**Comment by Jaxelr** [comment]
I'm requesting Scott to create a new Team so that my team can join as Service Owners. Thanks for all the feedback.

File: sdk/developer-signing/Azure.Developer.Signing/ci.yml
Comment (jsquire) [inline sdk/developer-signing/Azure.Developer.Signing/ci.yml:1]: I overlooked where this is.  This should be in /sdk/developer-signing, not under the package.

File: .github/CODEOWNERS
Comment (Jaxelr) [reply .github/CODEOWNERS]: Thanks, fixed.

File: .github/CODEOWNERS
Comment (jsquire) [inline .github/CODEOWNERS:182]:
```suggestion
```

The Azure SDK team does not own this package and will not be enhancing/maintaining it going forward, so we'll want just the service owners here.

File: sdk/developer-signing/Azure.Developer.Signing/tests/Samples/Sample_CreateCertificateProfileClient.cs
Comment (jsquire) [inline sdk/developer-signing/Azure.Developer.Signing/tests/Samples/Sample_CreateCertificateProfileClient.cs:11]: nit:  did we intend to mark this with `[Test]`?

File: sdk/developer-signing/tests.yml
Comment (jsquire) [inline sdk/developer-signing/tests.yml:7]:
```suggestion
    ServiceDirectory: developer-signing
```

File: sdk/developer-signing/Azure.Developer.Signing/tests/Samples/Sample_CreateCertificateProfileClient.cs
Comment (Jaxelr) [reply sdk/developer-signing/Azure.Developer.Signing/tests/Samples/Sample_CreateCertificateProfileClient.cs:11]: Yes, we should, thanks.

**Comment by Jaxelr** [comment]
@jsquire  It looks like the linter is complaining about a few entries I do not control, whats the guidance here? should I remove them?

**Comment by jsquire** [comment]
@Jaxelr :  The CI runs caught some things that will need to be fixed:

- Test recordings need to be made and pushed to the assets repository.    See [this doc](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/core/Azure.Core.TestFramework#recording).
- There are broken links in the README that still use the `developersigning` slug in their path

**Comment by Jaxelr** [comment]
Thanks, will take a look.

**Comment by Jaxelr** [comment]
hey @jsquire I made some updates, will you take a look whenever you have a moment? Thanks!

File: .github/CODEOWNERS
Comment (jsquire) [inline .github/CODEOWNERS:182]:
```suggestion
#PRLabel %Code Signing
/sdk/trustedsigning/                                               @jaxelr

# ServiceLabel: %Code Signing
# ServiceOwners:                                                   @jaxelr
```

Comment (jsquire) [reply .github/CODEOWNERS:182]: You need entries for both pull requests and support issues.   In this case, I'm assuming that you're the correct contact for both.

File: sdk/trustedsigning/Azure.Developer.Signing/README.md
Comment (jsquire) [inline sdk/trustedsigning/Azure.Developer.Signing/README.md:153]:
```suggestion
For more information on Azure SDK, please refer to [this website](https://azure.github.io/azure-sdk/)

![Impressions](https://azure-sdk-impressions.azurewebsites.net/api/impressions/azure-sdk-for-net%2Fsdk%2Ftrustedsigning%2FAzure.Developer.Signing%2FREADME.png)
```

Comment (jsquire) [reply sdk/trustedsigning/Azure.Developer.Signing/README.md:153]: You've got to keep the impression pixel.

File: sdk/trustedsigning/Azure.Developer.Signing/src/Azure.Developer.Signing.csproj
Comment (jsquire) [inline sdk/trustedsigning/Azure.Developer.Signing/src/Azure.Developer.Signing.csproj:3]: We should customize these; they're the metadata for the package that will appear on NuGet.

File: sdk/trustedsigning/Azure.Developer.Signing/src/GlobalSuppressions.cs
Comment (jsquire) [inline sdk/trustedsigning/Azure.Developer.Signing/src/GlobalSuppressions.cs:6]: Azure.Developer is an approved namespace.   We should be able to remove the ACZ0001 suppression here.

File: sdk/trustedsigning/Azure.Developer.Signing/tests/SessionRecords/DevSigningClientTests/GetCustomerCertificateEkuSuccessAsync.json
Comment (jsquire) [inline sdk/trustedsigning/Azure.Developer.Signing/tests/SessionRecords/DevSigningClientTests/GetCustomerCertificateEkuSuccessAsync.json:1]: The session records must be removed.   These should be using the assets system.   _(see: [docs](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core.TestFramework/README.md#recording))_

**Review by jsquire — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS
Comment (Jaxelr) [reply .github/CODEOWNERS:182]: Yes, I'll be the initial contact, will more than likely subscribe more individuals in the future. Thanks this has been fixed on the latest commit.

**Review by jsquire — APPROVED** [review]
LGTM.   Please let me know when you'd like me to merge.

File: sdk/trustedsigning/Azure.Developer.Signing/tests/SessionRecords/DevSigningClientTests/GetCustomerCertificateEkuSuccessAsync.json
Comment (Jaxelr) [reply sdk/trustedsigning/Azure.Developer.Signing/tests/SessionRecords/DevSigningClientTests/GetCustomerCertificateEkuSuccessAsync.json:1]: Removed session records.

File: sdk/trustedsigning/Azure.Developer.Signing/src/Azure.Developer.Signing.csproj
Comment (Jaxelr) [reply sdk/trustedsigning/Azure.Developer.Signing/src/Azure.Developer.Signing.csproj]: Good catch, thanks.

**Comment by Jaxelr** [comment]
Sure, we can merge @jsquire, Thanks.

cc: @Dawnwang17 @janielMartell

**Comment by jsquire** [comment]
> Sure, we can merge @jsquire, Thanks.
> 
> cc: @DawnWang17 @janielMartell

Once the live suite completes, if all is well, we'll merge.

**Comment by jsquire** [comment]
v

**Comment by jsquire** [comment]
@Jaxelr:  Pipeline runs have completed;  we'll need to resolve the failures before merging.   Looks like the test recordings didn't get pushed to the assets repo or the assets.json file wasn't added here.

**Comment by Jaxelr** [comment]
Ok, ill take a look, probably messed that up, thanks

**Comment by Jaxelr** [comment]
Ok, @jsquire  I think I got it this time, I also pushed the assets.json file:

image

Could be so kind as to review + retry the pipeline whenever you have a moment? Apologies for the toil.

**Comment by Jaxelr** [comment]
It looks like the tests are failing, I'll take a look at those.

File: assets.json
Comment (jsquire) [inline assets.json:1]: This is not in the right place.   This shouldn't be hanging out in the repro root.

**Review by jsquire — CHANGES_REQUESTED** [review]

File: assets.json
Comment (Jaxelr) [reply assets.json:1]: Woops, lol thanks for reviewing, I've been testing and accidentally committed this.

**Comment by Jaxelr** [comment]
hey @jsquire should I be concerned about that Build Analyze failure?

**Comment by jsquire** [comment]
> hey @jsquire should I be concerned about that Build Analyze failure?

Only if you want to merge.  😄     
All the CI and test pipelines must pass before final approval and merging.

**Comment by jsquire** [comment]
The codeowners linter failure isn't caused by this PR, it was the result of a path being removed yesterday.   Fixed by [43834](https://github.com/Azure/azure-sdk-for-net/pull/43834).

**Comment by jsquire** [comment]
@lirenhe, @chunyu3:

I could use an assist on the failure that seems to be consistently happening for the Analyze step during codegen validation here.:

```text
##[error]/mnt/vss/_work/1/.nuget/packages/microsoft.azure.autorest.csharp/3.0.0-beta.20240428.10/buildMultiTargeting/Microsoft.Azure.AutoRest.CSharp.targets(62,5): Error MSB3073: The command "pwsh -NoProfile -NonInteractive -executionpolicy Unrestricted -File /mnt/vss/_work/1/s/sdk/trustedsigning/Azure.Developer.Signing/src/../../../../eng/common/scripts/TypeSpec-Project-Sync.ps1 /mnt/vss/_work/1/s/sdk/trustedsigning/Azure.Developer.Signing/src/.. " exited with code 128.
```

Your thoughts would be appreciated.

**Comment by jsquire** [comment]
@Jaxelr:  Analyze error seems unrelated to these changes.   Can you try rebasing on `main` to see if that resolves the issue?

**Comment by jsquire** [comment]
@hallipr:  The error for the Analyze step here looks to be tooling/infrastructure-related.   Doesn't look like a rebase to `main` has resolved it.   I'd appreciate your thoughts.

**Review by jsquire — APPROVED** [review]
@Jaxelr:  When you're ready to merge, please mention me and I'll push the button.

**Comment by Jaxelr** [comment]
Thanks, let me take a look at those. @jsquire

**Comment by Jaxelr** [comment]
Hi @weshaggard any reason why you closed this PR? I dont mind creating a new request once we sort out the testing ordeal. Thanks.

**Comment by weshaggard** [comment]
@Jaxelr sorry I think this one was closed my mistake.

File: sdk/trustedsigning/Azure.Developer.Signing/README.md
Comment (dlemstra) [inline sdk/trustedsigning/Azure.Developer.Signing/README.md:7]: Are these links correct?

File: sdk/trustedsigning/Azure.Developer.Signing/README.md
Comment (dlemstra) [inline sdk/trustedsigning/Azure.Developer.Signing/README.md:26]: None of the other links have a trailing slash? Do we need it here?

File: sdk/trustedsigning/Azure.Developer.Signing/README.md
Comment (Jaxelr) [reply sdk/trustedsigning/Azure.Developer.Signing/README.md:7]: Theyre not, good catch :)

File: sdk/trustedsigning/Azure.Developer.Signing/README.md
Comment (Jaxelr) [reply sdk/trustedsigning/Azure.Developer.Signing/README.md:26]: No, its not needed.

File: sdk/trustedsigning/Azure.Developer.Signing/README.md
Comment (Jaxelr) [reply sdk/trustedsigning/Azure.Developer.Signing/README.md]: Fixed.

**Pull request closed without merging**
